File: src/java/array/TwoSum.java
package java.array;

import java.util.HashMap;

public class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hashmap = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (hashmap.containsKey(diff) && hashmap.get(diff) != i) {
                return new int[]{i, hashmap.get(diff)};
            }
            hashmap.put(nums[i], i);
        }
        return null;
    }

}

File: src/kotlin/array/TwoSum.kt
// O(n^2)
class TwoSum {

    private fun twoSum(nums: IntArray, target: Int): IntArray {
        for (i in nums.indices) {
            for (j in i + 1 until nums.size) {
                if (nums[i] + nums[j] == target) {
                    val arr = IntArray(2)
                    arr[0] = i
                    arr[1] = j
                    return arr
                }
            }
        }
        return IntArray(0)
    }

    fun main() {
        val arr = IntArray(5)
        arr[0] = 1
        arr[1] = 3
        arr[2] = 4
        arr[3] = 2
        arr[4] = 6
        print(twoSum(arr, 8).joinToString())
    }
}